Allow have multiple instances of BWEB

kant2002 opened the pull request:
The reliance on the static instance of the BWEM is convinient during development,
but when doing testing not so much.
Maybe multiple instance of BWEM right now don't possible,
but if my testing related changes would be accepted and would be any issues
I will go and beg, please, send PR to make testing of BWEM happens

This namespace contains functions which could be used for backward
compatibility with existing code. Just put
`using namespace BWEB::Utils`
at the beginning of the file where static Map methods was used
and replace all usages of `Map::overlapsBlocks` with just `overlapsBlocks`
Alternatively explicit using `BWEB::Utils::overlapBlocks` could be used.

One of the changes related to #11

N00byEdge commented on BWEB.h:
```diff
@@ -92,6 +93,7 @@ namespace BWEB
 		static Map* BWEBInstance;
 
 	public:
+		Map(BWEM::Map& map = BWEM::Map::Instance());
```
I'm not sure relying on that there is a static instance is the way to go here. I'd say that the default argument should be removed.

kant2002 replied:
I don't like it either, but I want to ensure 100% compatibility with existing code. I don't know is McRave only user, or somebody else use this lib? Do these people willing to change, so I go safe route. I'm always for having explicit dependencies, like seen from my PRs, But this looks safe default as temporary measure for me. So if everybody would like to remove default, I'm fully support that.

N00byEdge reviewed (changes requested):
BWEB should not rely on static BWEM instance.

N00byEdge commented:
Sorry messed up the first merge lol. I could of course make the commit myself if you would like.

kant2002 commented:
@N00byEdge Am I right that this is just should be rebased, to take into account that chrono was removed?

N00byEdge commented:
I merged the chrono changes into this PR as you can see by the 2 commits by me above by resolving the conflicts, it had to be done manually since it was the same code. A rebase only works if there are no conflicts.

kant2002 committed:
Allow have multiple instances of BWEB
The reliance on the static instance of the BWEM is convinient during development,
but when doing testing not so much.
Maybe multiple instance of BWEM right now don't possible,
but if my testing related changes would be accepted and would be any issues
I will go and beg, please, send PR to make testing of BWEM happens

This namespace contains functions which could be used for backward
compatibility with existing code. Just put
`using namespace BWEB::Utils`
at the beginning of the file where static Map methods was used
and replace all usages of `Map::overlapsBlocks` with just `overlapsBlocks`
Alternatively explicit using `BWEB::Utils::overlapBlocks` could be used.

One of the changes related to #11

Signed-off-by: Andrii Kurdiumov <kant2002@gmail.com>

N00byEdge committed:
Remove default argument for BWEM map
